src/main.cc

```diff
@@ -141,6 +141,7 @@ on_shutdown_f(va_list ap)
 	(void) ap;
 	trigger_fiber_run(&box_on_shutdown_trigger_list, NULL,
 			  on_shutdown_trigger_timeout);
+	say_logger_free();
```

Comment:
This change is not needed. Setting `log_boot` after `log_std` destroy is enough. I couldn't already reproduce the crash with just that change.

As for the trigger - I am rather against triggers here, because for the order of modules destruction we need more control. It needs to be resolved in scope of #7326 when we will need to resurrect all those `*_free()` calls.

As for when destroy, it shouldn't be done on shutdown, I think. Shutdown won't terminate the process, it only terminates the even loop. There will be more code executed afterwards which might need logging and other modules potentially too. Better deal with that in #7326.